test/e2e/auth/audit.go

```diff
@@ -580,45 +628,71 @@ var _ = SIGDescribe("Advanced Audit", func() {
-						responseObject: false,
+						level:             v1beta1.LevelMetadata,
+						stage:             v1beta1.StageResponseComplete,
+						requestURI:        fmt.Sprintf("/apis/%s/v1beta1/%s", crdNamespace, crdName),
+						verb:              "create",
+						code:              201,
+						user:              auditTestUser,
+						resource:          crdName,
+						requestObject:     false,
+						responseObject:    false,
+						authorizeDecision: "allow",
 					}, {
-						level:          v1beta1.LevelRequestResponse,
-						stage:          v1beta1.StageResponseComplete,
-						requestURI:     fmt.Sprintf("/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions/%s", crd.Name),
-						verb:           "delete",
-						code:           200,
-						user:           auditTestUser,
-						resource:       "customresourcedefinitions",
-						requestObject:  false,
-						responseObject: true,
+						level:             v1beta1.LevelRequestResponse,
+						stage:             v1beta1.StageResponseComplete,
+						requestURI:        fmt.Sprintf("/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions/%s", crd.Name),
+						verb:              "delete",
+						code:              200,
+						user:              auditTestUser,
+						resource:          "customresourcedefinitions",
+						requestObject:     false,
+						responseObject:    true,
+						authorizeDecision: "allow",
 					}, {
-						level:          v1beta1.LevelMetadata,
-						stage:          v1beta1.StageResponseComplete,
-						requestURI:     fmt.Sprintf("/apis/%s/v1beta1/%s/setup-instance", crdNamespace, crdName),
-						verb:           "delete",
-						code:           200,
-						user:           auditTestUser,
-						resource:       crdName,
-						requestObject:  false,
-						responseObject: false,
+						level:             v1beta1.LevelMetadata,
+						stage:             v1beta1.StageResponseComplete,
+						requestURI:        fmt.Sprintf("/apis/%s/v1beta1/%s/setup-instance", crdNamespace, crdName),
+						verb:              "delete",
+						code:              200,
+						user:              auditTestUser,
+						resource:          crdName,
+						requestObject:     false,
+						responseObject:    false,
+						authorizeDecision: "allow",
+					},
+				},
+			},
+			// get a pod with unauthorized user
+			{
+				func() {
+					_, err := anonymousClient.CoreV1().Pods(namespace).Get("another-audit-pod", metav1.GetOptions{})
+					expectForbidden(err)
+				},
+				[]auditEvent{
+					{
+						level:             v1beta1.LevelRequest,
+						stage:             v1beta1.StageResponseComplete,
+						requestURI:        fmt.Sprintf("/api/v1/namespaces/%s/pods/another-audit-pod", namespace),
+						verb:              "get",
+						code:              403,
+						user:              auditTestUser,
+						resource:          "pods",
+						namespace:         namespace,
+						requestObject:     false,
+						responseObject:    false,
+						authorizeDecision: "forbid",
```

Comment:
nit: almost all the tests use allow. Just make this default to allow in the tests, and then you only need to set it here.

Reply:
I think this needs a new function like newAuditEvent(), and this requires more modifications.